modules/ROOT/pages/migration-from-5x.adoc

```diff
@@ -3,88 +3,585 @@
+
+When the `editor.SetContent` event fires, any data in `content` has already been added to the {productname} editor instance and cannot be altered, making `content` essentially moot as an `editor.SetContent` property.
+
+To alter such data before it is added to the {productname} editor, use the `content` property available in the `BeforeSetContent` event.
+
+NOTE: Calling the `editor.SetContent` `content` property now returns an empty string.
+
+=== `fire` function renamed in all {productname} APIs
+
+The `fire()` function in
+
+* `tinymce.Editor`;
+* `tinymce.dom.EventUtils`;
+* `tinymce.dom.DOMUtils`;
+* `tinymce.util.Observable`; and
+* `tinymce.util.EventDispatcher`
+
+has been renamed to `dispatch()`.
+
+`fire` has been aliased to `dispatch` but has also been marked as _deprecated_.
+
+=== `new` keyword no longer used to instantiate plugins
+
+In {productname} 6.0, plugins (and other addons) are no longer initialised with the `new` keyword.
+
+As a consequence, the plugin initalizer value passed to `PluginManager.add` must be a function. It cannot be a class. (The plugin initializer value is the second argument passed to `PluginManager.add()`.)
+
+This change makes it safe to use both older functions and newer arrow functions when instantiating add-ons.
+
+=== PluginManager
+
+Previously, when loading a requested resource, the `PluginManager` passed a `success` or `failure` callback to report the status of the request. In {productname} 6, `PluginManager` reports this status by returning a `Promise`, with a value of either `resolve` or `reject`.
+
+=== redial
+
+Previously, when a component in a {productname} dialog box changed — for example, when a _next_ or _previous_ button was clicked — the entire dialog box was re-rendered.
+
+In {productname} 6.0, the `initialData` type in the dialog `redial` API uses object diffing for the passed spec. As a consequence, the API now only re-renders the portions of a dialog that have changed.
+
+=== ScriptLoader
+
+Previously, when loading a requested resource, `ScriptLoader` passed a `success` or `failure` callback to report the status of the request. In {productname} 6, `ScriptLoader` reports this status by returning a `Promise`, with a value of either `resolve` or `reject`.
+
+=== StyleSheetLoader
+
+Previously, when loading a requested resource, `StyleSheetLoader` passed a `success` or `failure` callback to report the status of the request. In {productname} 6, `StyleSheetLoader` reports this status by returning a `Promise`, with a value of either `resolve` or `reject`.
+
+=== ThemeManager
+
+Previously, when loading a requested resource, the `ThemeManager` passed a `success` or `failure` callback to report the status of the request. In {productname} 6, `ThemeManager` reports this status by returning a `Promise`, with a value of either `resolve` or `reject`.
+
+
+[[commands-and-queries]]
+== Commands and queries
+
+=== `element_format`
```

Comment:
This is an option, not a command, I'm pretty sure?

Reply:
Can confirm this is an option.

Reply:
Moved material from wrong section to right section.